### PR TITLE
privilege/privileges: migrate test-infra to testify

### LINKED ISSUE #26746: migrate test-infra to testify for `privilege/privileges` pkg

### PR DESCRIPTION
**Pull request by Ranxy**
Signed-off-by: Ranxy <huran@outlook.com>

### What problem does this PR solve?

Issue Number: close #26746 

### What is changed and how it works?

Migrate test-infra to testify for privilege/privileges pkg.

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by morgo — APPROVED** [review]

**Review by tisonkun — APPROVED** [review]

File: privilege/privileges/cache_test.go [hunk 1]
```diff
@@ -15,235 +15,244 @@ package privileges_test
...
-	c.Assert(p.DefaultRoles[0].User, Equals, "test_default_roles")
-	c.Assert(p.DefaultRoles[0].DefaultRoleHost, Equals, "localhost")
-	c.Assert(p.DefaultRoles[0].DefaultRoleUser, Equals, "r_1")
-	c.Assert(p.DefaultRoles[1].DefaultRoleHost, Equals, "localhost")
+	require.NoError(t, err)
+	require.Equal(t, `%`, p.DefaultRoles[0].Host)
+	require.Equal(t, "test_default_roles", p.DefaultRoles[0].User)
+	require.Equal(t, "localhost", p.DefaultRoles[0].DefaultRoleHost)
+	require.Equal(t, "r_1", p.DefaultRoles[0].DefaultRoleUser)
+	require.Equal(t, "localhost", p.DefaultRoles[0].DefaultRoleHost)
```
Comment (xhebox) [inline privilege/privileges/cache_test.go:187]: `p.DefaultRoles[1].DefaultRoleHost`

File: privilege/privileges/cache_test.go [hunk 2]
```diff
@@ -497,92 +511,96 @@ func (s *testCacheSuite) TestGlobalPrivValueRequireStr(c *C) {
...
-	c.Assert(none.RequireStr(), Equals, "NONE")
-	c.Assert(tls.RequireStr(), Equals, "SSL")
-	c.Assert(x509.RequireStr(), Equals, "X509")
-	c.Assert(spec.RequireStr(), Equals, "CIPHER 'c1' ISSUER 'i1' SUBJECT 's1'")
-	c.Assert(spec2.RequireStr(), Equals, "ISSUER 'i1' SUBJECT 's1'")
-	c.Assert(spec3.RequireStr(), Equals, "ISSUER 'i1'")
-	c.Assert(spec4.RequireStr(), Equals, "NONE")
+	require.Equal(t, "NONE", none.RequireStr())
+	require.Equal(t, "SSL", tls.RequireStr())
+	require.Equal(t, "X509", x509.RequireStr(), "X509")
```
Comment (xhebox) [inline privilege/privileges/cache_test.go:516]: Redundant `"X509"` comment

File: privilege/privileges/privileges_test.go [hunk 3]
```diff
@@ -25,502 +25,468 @@ import (
...
+	mustExec(t, se, `CREATE USER 'column'@'%'`)
+	mustExec(t, se, `CREATE TABLE column_table (a int, b int, c int)`)
+	mustExec(t, se, `GRANT Select(a),Update(a,b),Insert(c) ON test.column_table TO  'column'@'%'`)
 
 	pc := privilege.GetPrivilegeManager(se)
 	gs, err := pc.ShowGrants(se, &auth.UserIdentity{Username: "column", Hostname: "%"}, nil)
-	c.Assert(err, IsNil)
-	c.Assert(strings.Join(gs, " "), Equals, "GRANT USAGE ON *.* TO 'column'@'%' GRANT SELECT(a), INSERT(c), UPDATE(a, b) ON test.column_table TO 'column'@'%'")
+	require.NoError(t, err)
+	require.Equal(t, strings.Join(gs, " "), "GRANT USAGE ON *.* TO 'column'@'%' GRANT SELECT(a), INSERT(c), UPDATE(a, b) ON test.column_table TO 'column'@'%'")
```
Comment (xhebox) [inline privilege/privileges/privileges_test.go:383]: Reverted `expect` and `got`.

File: privilege/privileges/privileges_test.go [hunk 4]
```diff
@@ -843,330 +820,387 @@ func connectionState(issuer, subject pkix.Name, cipher uint16, opt ...func(c *x5
...
+	require.Error(t, err)
+	require.True(t, strings.Contains(err.Error(), "denied to user"))
 	_, err = se.ExecuteInternal(context.Background(), "drop table information_schema.tables")
-	c.Assert(strings.Contains(err.Error(), "denied to user"), IsTrue)
+	require.Error(t, err)
+	require.True(t, strings.Contains(err.Error(), "denied to user"))
 	_, err = se.ExecuteInternal(context.Background(), "update information_schema.tables set table_name = 'tst' where table_name = 'mysql'")
-	c.Assert(strings.Contains(err.Error(), "privilege check"), IsTrue)
+	require.Error(t, err)
+	require.True(t, strings.Contains(err.Error(), "privilege check"))
```
Comment (xhebox) [inline privilege/privileges/privileges_test.go:1136]: `require.Contains`?

File: privilege/privileges/privileges_test.go [hunk 5]
```diff
@@ -1740,80 +1856,87 @@ func (s *testPrivilegeSuite) TestSecurityEnhancedModeRestrictedUsers(c *C) {
...
-	c.Assert(pm.IsDynamicPrivilege("IAMdyNAMIC"), IsTrue)
-	c.Assert(len(privileges.GetDynamicPrivileges()), Equals, count+2)
+	require.False(t, pm.IsDynamicPrivilege("iAmdynamIC"))
+	require.Nil(t, privileges.RegisterDynamicPrivilege("IAMdynamic"))
+	require.True(t, pm.IsDynamicPrivilege("IAMdyNAMIC"))
+	require.Len(t, privileges.GetDynamicPrivileges(), count+2)
 
-	c.Assert(privileges.RegisterDynamicPrivilege("THIS_PRIVILEGE_NAME_IS_TOO_LONG_THE_MAX_IS_32_CHARS").Error(), Equals, "privilege name is longer than 32 characters")
-	c.Assert(pm.IsDynamicPrivilege("THIS_PRIVILEGE_NAME_IS_TOO_LONG_THE_MAX_IS_32_CHARS"), IsFalse)
+	require.Equal(t, privileges.RegisterDynamicPrivilege("THIS_PRIVILEGE_NAME_IS_TOO_LONG_THE_MAX_IS_32_CHARS").Error(), "privilege name is longer than 32 characters")
```
Comment (xhebox) [inline privilege/privileges/privileges_test.go:1904]: Reverted "expect" and "got".

**Review by xhebox — COMMENTED** [review]
Too long to review, maybe split next time.

File: testkit/result.go [hunk 6]
```diff
@@ -52,6 +53,23 @@ func Rows(args ...string) [][]interface{} {
 	return RowsWithSep(" ", args...)
 }
 
+// Sort sorts and return the result.
+func (res *Result) Sort() *Result {
+	sort.Slice(res.rows, func(i, j int) bool {
```
[the remark this answers is not in the document]
Comment (Ranxy) [reply testkit/result.go:58]: The type of `res.rows` is `[][]string`, which cannot be simply replaced by `sort.Strings(res.rows)`.

File: privilege/privileges/privileges_test.go (hunk 4 above)
Comment (Ranxy) [reply privilege/privileges/privileges_test.go:1136]: Maybe use `require.True(t, terror.ErrorEqual(err, core.ErrPrivilegeCheckFail))` instead?

File: privilege/privileges/cache_test.go (hunk 1 above)
Comment (Ranxy) [reply privilege/privileges/cache_test.go:187]: Will be fixed

File: privilege/privileges/cache_test.go (hunk 2 above)
Comment (Ranxy) [reply privilege/privileges/cache_test.go:516]: Will be fixed

File: privilege/privileges/privileges_test.go (hunk 3 above)
Comment (Ranxy) [reply privilege/privileges/privileges_test.go:383]: Will be fixed

File: privilege/privileges/privileges_test.go (hunk 5 above)
Comment (Ranxy) [reply privilege/privileges/privileges_test.go:1904]: Will be fixed

File: privilege/privileges/privileges_test.go (hunk 4 above)
Comment (xhebox) [reply privilege/privileges/privileges_test.go:1136]: It is even better!

File: testkit/result.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (xhebox) [reply testkit/result.go:58]: Resolved

**Review by xhebox — APPROVED** [review]

**Pull request merged**